src/main/java/org/zalando/failsafeactuator/aspect/FailsafeBreakerAspect.java

```diff
@@ -1,59 +1,75 @@
+ *
+ * <p>THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+ * BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 package org.zalando.failsafeactuator.aspect;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.concurrent.Callable;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import net.jodah.failsafe.CircuitBreaker;
 import net.jodah.failsafe.Failsafe;
 import org.aopalliance.intercept.MethodInterceptor;
 import org.aopalliance.intercept.MethodInvocation;
+import org.springframework.boot.actuate.metrics.CounterService;
+import org.zalando.failsafeactuator.metrics.CountingListener;
 import org.zalando.failsafeactuator.service.CircuitBreakerRegistry;
 
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
-import java.util.concurrent.Callable;
-
-
 @Slf4j
 @RequiredArgsConstructor
 public class FailsafeBreakerAspect implements MethodInterceptor {
 
   private final CircuitBreakerRegistry circuitBreakerRegistry;
+  private final CounterService counterService;
 
+  @Override
   public Object invoke(final MethodInvocation methodInvocation) throws Throwable {
     final Method method = methodInvocation.getMethod();
-    final org.zalando.failsafeactuator.aspect.Failsafe breaker = method.getAnnotation(org.zalando.failsafeactuator.aspect.Failsafe.class);
+    final org.zalando.failsafeactuator.aspect.Failsafe breaker =
+        method.getAnnotation(org.zalando.failsafeactuator.aspect.Failsafe.class);
+
 
     final CircuitBreaker circuitBreaker = circuitBreakerRegistry.getOrCreate(breaker.value());
     try {
       return Failsafe.with(circuitBreaker)
-                     .get(new Callable<Object>() {
-                       @Override
-                       public Object call() throws Exception {
-                         try {
-                           return methodInvocation.proceed();
-                         } catch (Throwable throwable) {
-                           throw new FailsafeBreakerWrappedException(throwable);
-                         }
-                       }
-                     });
-    } catch (RuntimeException e) {
+          .with(new CountingListener<>(counterService, breaker.value()))
+          .get(
+              new Callable<Object>() {
+                @Override
+                public Object call() throws Exception {
+                  try {
+                    return methodInvocation.proceed();
+                  } catch (final Throwable throwable) {
+                    throw new FailsafeBreakerWrappedException(throwable);
```

Comment:
Why is this custom exception type needed?

Reply:
@coders-kitchen should know. Depending on the outcome we should also tackle this on a separate issue as already mentioned before.

Reply:
I prefer to have explicit exceptions in cases like this, to make it clear what the cause in case anything unexpected happen.

Reply:
@coders-kitchen But you're deviating from the normal behaviour of failsafe. If a user would migrate from manually using failsafe to this annotation+advice, then he/she would experience a different behaviour. I believe it would be in the best interest of everybody if both ways are 100% compatible.